plugins/modules/website.ps1

```diff
@@ -1,27 +1,280 @@
+        }
 
-$module.Diff.before = $null
-$module.Diff.after = @{
-    name = $name
+        $site_parameters = @{
+            Name = $name
+            PhysicalPath = $physical_path
+        }
+
+        If ($application_pool) {
+            $site_parameters.ApplicationPool = $application_pool
+        }
+
+        If ($site_id) {
+            $site_parameters.ID = $site_id
+        }
+        # Fix for error "New-Item : Index was outside the bounds of the array."
+        # This is a bug in the New-WebSite commandlet. Apparently there must be at least one site configured in IIS otherwise New-WebSite crashes.
+        # For more details, see http://stackoverflow.com/questions/3573889/ps-c-new-website-blah-throws-index-was-outside-the-bounds-of-the-array
+        $sites_list = Get-ChildItem -LiteralPath IIS:\sites
+        if ($null -eq $sites_list) {
+            if ($site_id) {
+                $site_parameters.ID = $site_id
+            }
+            else {
+                $site_parameters.ID = 1
+            }
+        }
+        if ( -not $check_mode) {
+            $site = New-Website @site_parameters -Force
+        }
+        # Verify that initial site has no binding
+        Get-WebBinding -Name $site.Name | Remove-WebBinding -WhatIf:$check_mode
+        $module.Result.changed = $true
+    }
+    # Remove site
+    If ($state -eq 'absent' -and $site) {
+        $site = Remove-Website -Name $name -WhatIf:$check_mode
+        $module.Result.changed = $true
+    }
+    $site = Get-Website | Where-Object { $_.Name -eq $name }
+    If ($site) {
+        # Change Physical Path if needed
+        if ($physical_path) {
+            If (-not (Test-Path -LiteralPath $physical_path)) {
+                $module.FailJson("specified folder must already exist: physical_path $($_.Exception.Message)")
+            }
+
+            $folder = Get-Item -LiteralPath $physical_path
+            If ($folder.FullName -ne $site.PhysicalPath) {
+                Set-ItemProperty -LiteralPath "IIS:\Sites\$($site.Name)" -name physicalPath -value $folder.FullName -WhatIf:$check_mode
+                $module.Result.changed = $true
+            }
+        }
+        # Change Application Pool if needed
+        if ($application_pool) {
+            If ($application_pool -ne $site.applicationPool) {
+                Set-ItemProperty -LiteralPath "IIS:\Sites\$($site.Name)" -name applicationPool -value $application_pool -WhatIf:$check_mode
+                $module.Result.changed = $true
+            }
+        }
+        # Add Remove or Set bindings if needed
+        if ($bindings) {
+            $site_bindings = (Get-ItemProperty -LiteralPath "IIS:\Sites\$($site.Name)").Bindings.Collection
+            $user_bindings = if ($Module.Params.bindings.add) { $Module.Params.bindings.add }
+            elseif ($Module.Params.bindings.remove) { $Module.Params.bindings.remove }
+            elseif ($Module.Params.bindings.set) { $Module.Params.bindings.set }
```

Comment:
This logic doesn't sound right to me. It seems like you want to check if `set` is present and use if, otherwise you need to check both add and remove together to add and remove accordingly.